### DISCUSSION TITLE
Event Widget

### DISCUSSION THREAD

**Discussion by StingX2** [RetroAchievements/RAWeb#1098] [Ideas]
A discussed idea many times in Staff. When this widget was originally added by Scott it was by my request as AotW was the ONLY event on the site. Now we have enough yearly and seasonal events throwing all the attention on this one event doesn't make sense, even with a pending events team.

![chrome_78qF2MMTE0](https://user-images.githubusercontent.com/70493569/185711680-cf1e6d0f-71ec-4c4b-b3bf-a4ca06c60737.png)

Events are hard to explain but statistically getting users started is the hook for many success stories. I can point to users that have joined discord or even just become involved past events (new devs, mods, code reviewers, artists, writers and more oh my!) period because of both Daily Distractions, Unwanted, Ra Roulette, RA Olympics, Challenge League and LOTM. 

We had discussed waiting til V2 to properly address this but with that not happening heres my proposal

A. A carousal widget on the front page that rotates a highlight from each event. This may not work with the daily events but the weekly events it would

B. Dedicate a week to each event in a rotation. In this sense even the daily distractions esque daily event has a shot at being centered on the front page

Marketing in Community news, front page carousel news, and this widget are all important. People are reluctant to try new things, its why new products are shoved down our throats via advertisement until we get curious enough to try.

Events are the lifeblood of RA because they don't just promote community, they are the gateway drug to involvement.

**Comment by bendyhuman** [comment]
I want to really stress your last point, that events "are the gateway drug to involvement", because I think that's what's really key to this whole idea. There are so many ways of getting involved in RA, but they all involve overcoming that first step beyond just earning achievements all alone. I can understand why many users may not want to engage with Discord, or may not be interested in achievement development, but events are such a natural extension of why we join the site in the first place - to play classic games in new ways and be rewarded for it! In my experience with event hosting, one of my biggest surprises was how many participants I didn't recognize as regulars in the community. These site-only users found my event, and in many cases that was their first step toward getting involved in other ways.

I agree that discoverability is definitely an issue with site events. With the way they're hidden away in the Competitions section of the forums, I imagine many non-Discord users don't even realize that they exist, unless they happen to catch the start of a new event in the front page news carousel. These updates come and go rather quickly, though, and especially aren't ideal for the kinds of ongoing events that you can join at any time. I don't think it's far fetched to believe that there is a non-negligible subset of users who are aware of Achievement of the Week due to its permanent front page placement, but have no idea that other events also exist.

As for the format such an events widget could take, either of your ideas could work just fine honestly, but I lean toward Option A. In terms of site maintenance, I like the idea of it only needing to be updated once a week, and I imagine this would be part of the duties of the events team. While Option A could include the actual relevant achievements for events such as AotW or RAWR, I think it would be just fine for more complex events like Daily DistRActions or The Unwanted to just have a brief description of the event with a link to the forum thread. While Option B would give individual events more time in the spotlight, it sort of gets back to the issue with having just the AotW widget. I think it would be more beneficial to clearly make users aware of all currently running events.

**Comment by KingS1zzle** [comment]
I agree with the carousal widget on the front page that rotates a highlight from each event.

This could just be larger or the same size as the Achievement of the Week. Just like the News carousal this would allow users to see updated active events and scroll through them as they wish. 
Even if its not even just Achievements, you could put a brief description underneath the title of the events.

Not sure if it would be the best idea, but you could reduce the amount of Event news posting if this carousal widget existed. Maybe only post on the news if a new event starts? The carousal could show the newest updated event first, while still including the AOTW.

**Comment by Xeinok-Pals** [comment]
I strongly dislike carousels - I feel like they needlessly hide content and are a bit out of date. I would greatly prefer some kind of event board: 
![image](https://user-images.githubusercontent.com/65511279/223209952-1629bc53-8e54-4939-9ec7-b01924a0698d.png)

It's important to succinctly explain to a wide range of users exactly what each event is and to demonstrate just how easy they are to join. A little section which shows potential rewards might be very enticing as well.

**Comment by wescopeland** [comment]
I'd like to resurface this conversation to figure out how we can work together to tackle this. Preferably without a carousel.

Some initial potential ideas:
* Pick a random event and show it in the widget with a link somewhere to all other events.
* Show a different event randomly on each day.

We need to also determine which events will be shown, as it would probably make sense for them to have tiny custom-made components similar to the current AOTW component.

**Reply by StingX2** [reply]
Random widget sounds cool just to need to make sure events that are highlighted arent like cream of the crop which has had closed signups for a month+

or

If its a preplanned event each day we'd want to time them around event updates
Saturday: console wars
Sunday: rawr
Monday: aotw
etcetc

**Reply by bendyhuman** [reply]
It doesn't need to be a carousel specifically, those are good suggestions. I think at this point getting SOMETHING implemented is more important than the specific details of how it should be presented. I personally like the idea of it highlighting a random event each day if that's viable. The events team can prepare a short blurb for each event that's currently ongoing/open for signups, and we can give the web team a poke whenever something needs to be added or removed, which really isn't that often at all. The link to "all events" could just link to the competitions section of the forums, or we could create a new forum topic that contains links to all other events with a brief description of each.

**Reply by wescopeland** [reply]
My initial thinking is we should first identify the "evergreen" events (such as AOTW) that are running indefinitely. We can build components for those first because they are the lowest common denominator in terms of complexity.

Afterwards, we can graduate to events that have fixed schedules/timeframes/end dates, as their logic is going to have a little bit more complexity. For evergreen events, I think they are:

* AOTW
* The Unwanted
* RA Roulette
* Peak Streak

I think some of these "end" at the end of the year, will they also be restarting in Jan 2024? I'm looking for the events that are kind of like 24/7 diners in that they never really close. This list may be inaccurate or incorrect, I could use some help on putting it together.

**Reply by bendyhuman** [reply]
Here are all current events and their schedules, in no particular order:

Peak Streak - Daily "evergreen" event, never introduces anything new.

Challenge League The Top 100 - Runs through all of 2023, with new challenges on the 1st of each month. Each year typically has a new Challenge League with a unique setup.

Cream of the Crop - Currently running event but signups are closed, so this doesn't need to be highlighted.

Simple Monthly Master - No fixed end date; post your masteries whenever, and stats are posted each month.

Achievement of the Week - Featured achievements rotate weekly (Monday 00:00 UTC) and monthly (first Monday of the month 00:00 UTC). Each year features a new forum topic with some changes and a fresh leaderboard/badges. Also features occasional seasonal sub-events that would be nice to highlight (ex: https://retroachievements.org/viewtopic.php?t=21062).

Console Wars - Weekly event, resets Satudays 00:00 UTC. End date is flexible depending on how the event plays out. Will likely have a future season with a new forum topic.

RA Roulette (RAWR) - Weekly event, resets Saturdays 11am EST. Runs through Feb 3rd 2024, will likely continue yearly.

The Unwanted - Evergreen event with ongoing player stats that updates monthly with new featured games and a new forum topic on the 15th of each month. Main thread available here: https://retroachievements.org/viewtopic.php?t=14787

House Cleaning - Monthly-ish with a new thread each time. Inconsistent schedule so far, I think they're aiming for the 2nd of each month moving forward? Current month runs through June 2nd.

Draw My Pokemon - Community submission phase is over, no need to highlight this one.

Leaderboard of the Month - Updates on the 1st and 16th of each month with an ongoing scoreboard.
